packages/arb-rpc-node/cmd/arb-node/arb-node.go

```diff
@@ -609,50 +625,81 @@ func startValidator(
-	chainStateFile, err := os.Open(chainStatePath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return nil, errors.Wrap(err, "failed to open chainState.json")
+	if config.Validator.WalletAddress != "" {
+		if !ethcommon.IsHexAddress(config.Validator.WalletAddress) {
+			log.Error().Str("address", config.Validator.WalletAddress).Msg("invalid validator smart contract wallet")
+			return nil, errors.New("invalid validator smart contract wallet")
 		}
+		chainState.ValidatorWallet = config.Validator.WalletAddress
 	} else {
-		chainStateData, err := ioutil.ReadAll(chainStateFile)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to read chain state")
+		if config.Validator.WalletAddressFilename == "" {
+			config.Validator.WalletAddressFilename = "chainState.json"
 		}
-		err = json.Unmarshal(chainStateData, &chainState)
+
+		config.Validator.WalletAddressFilename = path.Join(config.Persistent.Chain, config.Validator.WalletAddressFilename)
+		chainStateFile, err := os.Open(config.Validator.WalletAddressFilename)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to unmarshal chain state")
+			// If file doesn't exist yet, will be created when needed
+			if !os.IsNotExist(err) {
+				return nil, errors.Wrap(err, "failed to open chainState file: "+config.Validator.WalletAddressFilename)
+			}
+		} else {
+			chainStateData, err := ioutil.ReadAll(chainStateFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to read chain state")
+			}
+			err = json.Unmarshal(chainStateData, &chainState)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to unmarshal chain state")
+			}
 		}
 	}
 
 	var valAuth transactauth.TransactAuth
+	var err error
 	if len(walletConfig.Fireblocks.SSLKey) > 0 {
 		valAuth, _, err = transactauth.NewFireblocksTransactAuthAdvanced(ctx, l1Client, auth, walletConfig, false)
 	} else {
 		valAuth, err = transactauth.NewTransactAuthAdvanced(ctx, l1Client, auth, false)
 	}
 	if err != nil {
-		return nil, errors.Wrap(err, "error creating connecting to chain")
+		return nil, errors.Wrap(err, "error connecting to l1 chain")
 	}
 	var validatorAddress *ethcommon.Address
 	if chainState.ValidatorWallet != "" {
+		log.Info().Str("address", chainState.ValidatorWallet).Msg("validator using smart contract wallet")
 		addr := ethcommon.HexToAddress(chainState.ValidatorWallet)
 		validatorAddress = &addr
+	} else {
+		log.Info().Msg("validator smart contract wallet creation delayed until needed")
 	}
 	onValidatorWalletCreated := func(addr ethcommon.Address) {
 		chainState.ValidatorWallet = addr.String()
 		newChainStateData, err := json.Marshal(chainState)
 		if err != nil {
 			log.Warn().Err(err).Msg("failed to marshal chain state")
-		} else if err := ioutil.WriteFile(chainStatePath, newChainStateData, 0644); err != nil {
+		} else if err := ioutil.WriteFile(config.Validator.WalletAddressFilename, newChainStateData, 0644); err != nil {
```

Comment:
We should confirm this will only be true is `config.Validator.WalletAddress` wasn't set since otherwise this path could be invalid

Reply:
confirmed, `CreateValidatorWallet` not called if address provided.

Reply:
updated to just print error if file saved when WalletAddress provided